sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala

```diff
@@ -570,4 +596,22 @@ class JoinHintSuite extends PlanTest with SharedSparkSession with AdaptiveSparkP
       assert(joinHints == expectedHints)
     }
   }
+
+  test("SPARK-32220: Non Cartesian Product Join Result Correct with SHUFFLE_REPLICATE_NL hint") {
```

Comment:
So, is this a correctness issue, @AngersZhuuuu ?

Reply:
Yea, I think so. Nice catch, @AngersZhuuuu

Reply:
Yea,  when I try new join hint, I found this result is non-correct.